lambda/events_id/get/index.js

```diff
@@ -2,104 +2,121 @@
-        const data = await documentClient.query(params).promise();
-        responseBody = data;
-
-        // Checking if the query returned with an event or found no matches
-        if(responseBody.Count == 0) {
-            //setting responseBody and statusCode on no results returned from DynamoDB
-            responseBody = 'Event not found';
-            statusCode = 404;
-            
-            // Log for no event found
-            console.log("Query did not find a matching event with the event_id provided. Status code: ", statusCode);
-        } else {
-            //setting responseBody and statusCode on successful return of event from DynamoDB
-            responseBody = JSON.stringify(responseBody.Items);
-            statusCode = 200;
-            
-            // Log for event found
-            console.log("Event found, status code: ", statusCode);
-        }
-        
-    } catch(err) {
-        responseBody = 'Failure to connect or retrieve from the database';
-        statusCode = 500;
-        
-        // Log for error with DynamoDB query
-        console.error("Unsuccessful connection to DynamoDB. Status code: " + statusCode);
-        console.error("Error: " + err);
-    }//end of catch
+    //Log calling API type and endpoint
+    console.log("API-Endpoint called: " + httpMethod + " " + event.context['resource-path']);
+    
+    //Log calling IP
+    console.log("Calling-IP: " + event.params.header['X-Forwarded-For']);
 
     // Create response to return with statusCode, headers, and responseBody. 
     const response = {
         statusCode: statusCode,
-        headers : {
-            "Access-Control-Allow-Origin" : "*"
-        },
-        body: responseBody
+        body: body
     };
+    
+    // validate input parameter as uuid version 5, when fails a 400 response must be given
+    if(!isUUID.anyNonNil(eventId)) {
+    	console.log("ERROR: uuid in not uuid/v4");
+    	response.statusCode = 400;
+		callback(JSON.stringify(response), null);
+	
+    }
+    else {
+
+        // parameters for DynamoDB request. 
+        const params = {
+            TableName: 'ak-prod-events-dynamo',
+            KeyConditionExpression: "#event_id = :event_id",
+            // Specify which attribute to search for
+            ExpressionAttributeNames:{
+                "#event_id": "event_id"
+            },
+            // Specify which attribute value {event id} to search for
+            ExpressionAttributeValues: {
+                ":event_id": eventId
+            },
+            // ProjectionExpression is for the request attributes of events in table 
+            ProjectionExpression: "event_id, user_name, activity_type, description, organization_name, location_name, location_address, contact_name, contact_phone, contact_email, start_date_time, end_date_time, frequency, cost, picture_url, min_age, max_age, disability_types, show_status, approver"
```

Comment:
The projection expression appears to have some incorrect parameters.
- organization_name should be org_name
- show_status should be event_status
- missing inclusive_event

Reply:
Also missing event_link

Reply:
fixed